library/src/updater.rs

```diff
@@ -131,41 +163,20 @@ fn check_hash(path: &Path, expected_string: &str) -> anyhow::Result<bool> {
 // And also avoid (for now) dealing with inflating patches on iOS.
 #[cfg(any(target_os = "android", test))]
 fn prepare_for_install(
-    config: &ResolvedConfig,
+    config: &UpdateConfig,
     download_path: &Path,
     output_path: &Path,
 ) -> anyhow::Result<()> {
-    use crate::android::{app_data_dir_from_libapp_path, open_base_lib};
-
-    // FIXME: This makes the assumption that the last path provided is the full
-    // path to the libapp.so file.  This is true for the current engine, but
-    // may not be true in the future.  Better would be for the engine to
-    // pass us the path to the base.apk.
-    // https://github.com/shorebirdtech/shorebird/issues/283
-    // This is where the paths are set today:
-    // First path is "libapp.so" (for dlopen), second is a full path:
-    // https://github.com/flutter/engine/blob/a7c9cc58a71c5850be0215ab1997db92cc5e8d3e/shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java#L264
-    // Which is composed from nativeLibraryDir:
-    // https://developer.android.com/reference/android/content/pm/ApplicationInfo#nativeLibraryDir
-    let full_libapp_path = config
-        .original_libapp_paths
-        .last()
-        .context("No libapp paths")?;
-    // We could probably use sourceDir instead?
-    // https://developer.android.com/reference/android/content/pm/ApplicationInfo#sourceDir
-    // and splitSourceDirs (api 21+)
-    // https://developer.android.com/reference/android/content/pm/ApplicationInfo#splitSourceDirs
-    debug!("Finding apk from: {:?}", full_libapp_path);
-    let app_dir = app_data_dir_from_libapp_path(full_libapp_path)?;
+    // We abuse libapp_path to actually be the path to the data dir for now.
+    let app_dir = &config.libapp_path;
```

Comment:
Updating the comment to explain why this is abuse and what the correct behavior would be would be helpful